internal/services/cdn/cdn_frontdoor_custom_domain_association_resource.go

```diff
@@ -0,0 +1,237 @@
+
+	id, err := parse.FrontDoorCustomDomainAssociationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.ProfileName, id.AssociationName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	return nil
+}
+
+func resourceCdnFrontDoorCustomDomainAssociationUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Cdn.FrontDoorCustomDomainsClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	if d.HasChange("cdn_frontdoor_route_ids") {
+		cdId, err := parse.FrontDoorCustomDomainIDInsensitively(d.Get("cdn_frontdoor_custom_domain_id").(string))
+		if err != nil {
+			return err
+		}
+
+		id := parse.NewFrontDoorCustomDomainAssociationID(cdId.SubscriptionId, cdId.ResourceGroup, cdId.ProfileName, cdId.CustomDomainName)
+
+		existing, err := client.Get(ctx, cdId.ResourceGroup, cdId.ProfileName, cdId.CustomDomainName)
+		if err != nil {
+			if utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("updating %s: %s was not found", id, cdId)
+			}
+
+			return fmt.Errorf("updating %s: %+v", id, err)
+		}
+
+		// make sure the routes exist and are valid for this custom domain...
+		routes, err := flattenRoutes(d, meta, cdId)
+		if err != nil {
+			return fmt.Errorf("updating %s: %+v", id, err)
+		}
+
+		d.Set("cdn_frontdoor_route_ids", routes)
+	}
+
+	return resourceCdnFrontDoorCustomDomainAssociationRead(d, meta)
+}
+
+func resourceCdnFrontDoorCustomDomainAssociationDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	// since you are deleting the resource you cannot grab the value from the config
+	// because it will be empty, you have to get it from the states old value...
+	oCdId, _ := d.GetChange("cdn_frontdoor_custom_domain_id")
+
+	cdId, err := parse.FrontDoorCustomDomainIDInsensitively(oCdId.(string))
+	if err != nil {
+		return err
+	}
+
+	id, err := parse.FrontDoorCustomDomainAssociationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	oRids, _ := d.GetChange("cdn_frontdoor_route_ids")
+	oR := oRids.([]interface{})
+
+	v, _, err := routesInsensitively(oR)
+	if err != nil {
+		return err
+	}
+
+	if len(*v) != 0 {
+		if err := removeCustomDomainAssociationFromRoutes(d, meta, v, cdId); err != nil {
+			return fmt.Errorf("deleting %s: %+v", id, err)
+		}
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func flattenRoutes(d *pluginsdk.ResourceData, meta interface{}, id *parse.FrontDoorCustomDomainId) ([]interface{}, error) {
+	out := make([]interface{}, 0)
+	o, n := d.GetChange("cdn_frontdoor_route_ids")
+	oRoutes := o.([]interface{})
+	nRoutes := n.([]interface{})
+
+	if len(nRoutes) == 0 || nRoutes == nil || id == nil {
+		return out, nil
+	}
+
+	oIds, _, err := routesInsensitively(oRoutes)
+	if err != nil {
+		return out, err
+	}
+
+	nIds, result, err := routesInsensitively(nRoutes)
```

Comment:
both of these are coming from the config and so can be parsed case-sensitively

Reply:
Fixed.